File: web/plugins/ceil.js
import { WebPlugin } from "../engine/plugin.js";

export class CeilPlugin extends WebPlugin {
    constructor() {
        super("切り上げ");
    }

    async execute(line, context) {
        const expression = line.replace(/^切り上げ\s*/, "");
        const value = Number(
            context.parser.parseValue(expression, context.runtime)
        );

        context.runtime.write(Math.ceil(value));
    }
}

File: pc/plugins/ceil.py
import re
from core.plugin import Plugin


class CeilPlugin(Plugin):
    def translate(self, line):
        match = re.fullmatch(r'切り上げ\s+(.+)', line)
        if not match:
            return None

        value = match.group(1)
        return f"print(__import__('math').ceil({value}))"